core/script_debugger_remote.cpp

```diff
@@ -505,7 +505,10 @@ void ScriptDebuggerRemote::line_poll() {
 
 	// Checking for stale watches
```

Comment:
This whole thing should probably not happen on each line, but rather use the if statement above or some predefined frequency.